### PR TITLE
Hole context generation and unit tests

### PR DESCRIPTION
**Pull request by ruiz-m**
This adds an extension to the dhexp_of_uexp function to return the hole context generated by the given uexp.
It also adds unit tests for the dhexp_of_uexp function using the Tezt library.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -51,7 +51,7 @@ repl:
 
 test:
 	dune build @src/fmt --auto-promote || true
-	dune exec src/hazeltest/hazeltest.exe -- --regression-dir src/hazeltest/regressions
+	dune exec src/haz3ltest/haz3ltest.exe -- --regression-dir src/haz3ltest/regressions
```
Comment (cyrus-) [inline Makefile]: can you add a blurb to the README file for how to run tests (I removed it recently from dev since it was out of date)

**Review by cyrus- — CHANGES_REQUESTED** [review]
I'm getting an error when I run `make test` -- is that the right way to run the tests or am I missing something?

File: README.md [hunk 2]
```diff
@@ -180,6 +180,11 @@ for a type named something else like `q`, it will be `show_q`.
 #### Debug Mode
 If Hazel is hanging on load or when you perform certain actions, you can load into Debug Mode by appending `#debug` to the URL and reloading. From there, you have some buttons that will change settings or reset local storage. Refresh without the `#debug` flag and hopefully you can resolve the situation from there.
 
+#### Testing
+You can run all of the unit tests located in `src/hazelcore/test` by running `make test`.
```
Comment (cyrus-) [inline README.md]: the tests are in `src/haz3ltest` not `src/hazelcore/test`

File: README.md [hunk 3]
```diff
@@ -180,6 +180,11 @@ for a type named something else like `q`, it will be `show_q`.
...
+
+Unit tests are written using the Tezt library.
```
Comment (cyrus-) [inline README.md]: Make the words "Tezt library" a link to the documentation for the library.

File: src/haz3lcore/dynamics/Delta.rei [hunk 4]
```diff
@@ -4,6 +4,19 @@ type hole_sort =
   | PatternHole;
 
 [@deriving sexp]
-type t = MetaVarMap.t((hole_sort, Typ.t, VarCtx.t));
+type val_ty = (hole_sort, Typ.t, Ctx.t);
+
+[@deriving sexp]
+type t = MetaVarMap.t((hole_sort, Typ.t, Ctx.t));
```
Comment (cyrus-) [inline src/haz3lcore/dynamics/Delta.rei]: should take `val_ty` as argument

Comment (cyrus-) [inline src/haz3lcore/dynamics/Delta.rei]: should take `val_ty` as argument

**Comment by cyrus-** [comment]
@ruiz-m can you take a look at the small changes above

**Review by cyrus- — CHANGES_REQUESTED** [review]

**Comment by ruiz-m** [comment]
I have some concerns. Since hole closure environment is not used in type assignment like in the POPL 2019 paper, Ctx.t in Deta.t is never used in the type assignment module.
Also, after merging the adt-defs-after branch with the syntax checker I noticed that the structure of the elaborator.re file was different. I am worried that there might be some merging issues in the future.

**Comment by cyrus-** [comment]
Yeah I think it might be helpful to rebase this on adt-defs-after and resolve the conflicts there. Also, I was thinking that Delta.t might be derivable directly from the info map?

**Comment by cyrus-** [comment]
Closing this PR in favor of separate PRs for test framework and hole context stuff

**Pull request closed without merging**